[reland] delete frontend.cpp

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67401
* __->__ #67400

reland of https://github.com/pytorch/pytorch/pull/66581
c10d/frontend.cpp was originally proposed to introduce pure C++ API and use TorcBind to share python level API with TorchScript. This is no longer needed, so delete this to reduce code redundancy.

Differential Revision: [D31979270](https://our.internmc.facebook.com/intern/diff/D31979270/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31979270/)!

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

Commit:
[BE] delete frontend.cpp

c10d/frontend.cpp was originally proposed to introduce pure C++ API and use TorcBind to share python level API with TorchScript. This is no longer needed, so delete this to reduce code redundancy.
ghstack-source-id: 140545797

Differential Revision: [D31979270](https://our.internmc.facebook.com/intern/diff/D31979270/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31979270/)!

[ghstack-poisoned]

mrshenli reviewed (approved):
Stamp to unblock. Please wait for the tests added by "ci/master" tag.

Commit:
Update on "[reland] delete frontend.cpp"


reland of https://github.com/pytorch/pytorch/pull/66581
c10d/frontend.cpp was originally proposed to introduce pure C++ API and use TorcBind to share python level API with TorchScript. This is no longer needed, so delete this to reduce code redundancy.

Differential Revision: [D31979270](https://our.internmc.facebook.com/intern/diff/D31979270/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31979270/)!

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

Commit:
Update on "[reland] delete frontend.cpp"


reland of https://github.com/pytorch/pytorch/pull/66581
c10d/frontend.cpp was originally proposed to introduce pure C++ API and use TorcBind to share python level API with TorchScript. This is no longer needed, so delete this to reduce code redundancy.

Differential Revision: [D31979270](https://our.internmc.facebook.com/intern/diff/D31979270/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31979270/)!

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

Commit:
Update on "[reland] delete frontend.cpp"


reland of https://github.com/pytorch/pytorch/pull/66581
c10d/frontend.cpp was originally proposed to introduce pure C++ API and use TorcBind to share python level API with TorchScript. This is no longer needed, so delete this to reduce code redundancy.

Differential Revision: [D31979270](https://our.internmc.facebook.com/intern/diff/D31979270/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31979270/)!

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

Commit:
Update on "[reland] delete frontend.cpp"


reland of https://github.com/pytorch/pytorch/pull/66581
c10d/frontend.cpp was originally proposed to introduce pure C++ API and use TorcBind to share python level API with TorchScript. This is no longer needed, so delete this to reduce code redundancy.

Differential Revision: [D31979270](https://our.internmc.facebook.com/intern/diff/D31979270/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31979270/)!

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]
